api/src/main/java/org/openmrs/module/pihcore/apploader/apps/patientregistration/SectionsLiberia.java

```diff
@@ -34,6 +36,33 @@ public void addSections(RegistrationAppConfig c) {
         }
     }
 
+    @Override
+    public Section getContactInfoSection() {
+        // build the default contact info section and retrieve the address question
+        Section contactInfoSection = super.getContactInfoSection();
+
+        Question homelessQuestion = new Question();
+        homelessQuestion.setHeader("pihcore.homeless");
+        homelessQuestion.setLegend("pihcore.homeless");
+
+        // add the homeless section to it.
+        Field field = new Field();
+        field.setFormFieldName("homeless");
+        field.setType("personAttribute");
+        field.setLabel("pihcore.homeless");
+        field.setUuid(Metadata.getHomelessAttributeType().getUuid());
+
+        CheckboxWidget w = new CheckboxWidget();
+        w.getConfig().setValue("true");
+        field.setWidget(toObjectNode(w));
+        homelessQuestion.addField(field);
+
+        // insert after the "address" question (assumption: address is first question)
+        contactInfoSection.getQuestions().add(1, homelessQuestion);
+
+        return contactInfoSection;
+    }
+
```

Comment:
This all looks fine, I guess, but FWIW, I do not think modeling this as a Person Attribute is a good idea.  Person Attributes are best used for things that are (ideally unchanging) attributes of a Person.  "Is Test Patient".  "Eye Color".  "Birthplace".  These things are set once, just like known attributes like Birthdate, and never (or rarely) changed.

Homeless is (hopefully) a temporary state that a person is in.  If they are homeless at the time of registration, we should record that as an Obs on their registration encounter.  If we want to see if this has changed over time, we can collect additional answers to this question in subsequent encounters.  This would seem a more appropriate way to model this.

@mogoodrich / @lnball ?